### DISCUSSION TITLE
Not able to view folders

### DISCUSSION THREAD

**Discussion by foxhoundv** [nextcloud/all-in-one#1705] [Questions] [answered]
### Steps to reproduce
1. Update to Nextcloud Hub 3 (25.0.2) Nextcloud AIO v4.0.1 from v3.???(I don't remember the rest of the numbers)
2. Remove "AppOrder" app because it was deprecated (Problem was present at this time so I did this hoping it would fix)
3. Install "Custom Menu" app to replace previous one 

### Expected behavior 
1. I should be able to click on "files" in the menu and see all my folders and files.

### Actual behavior 
1. 
![image](https://user-images.githubusercontent.com/16512250/211355809-eba335c6-2d31-488c-b162-64ef92158972.png)
2. 
![image](https://user-images.githubusercontent.com/16512250/211355939-3e5ddf4a-bc77-4e06-8fd1-eabd96e45315.png)

### Host OS 
Ubuntu 20.04.5 LTS

#### Nextcloud AIO version 
v4.0.1

#### Current channel 
latest

**Comment by szaimen** [comment]
Hm... do you have something in your server logs?

**Comment by foxhoundv** [comment]
Through the AIO interface?

**Comment by szaimen** [comment]
> Through the AIO interface?

no, from here: `https://yourdomain.com/settings/admin/logging`

**Comment by foxhoundv** [comment]
Can I message you through reddit because it has some sensitive stuff?

**Comment by szaimen** [comment]
> Can I message you through reddit because it has some sensitive stuff?

you can send it to my mail-address...

**Comment by szaimen** [comment]
Also please post your browser logs:

Open your console, reload your page and/or do the action leading to this issue and copy/paste the log in this thread.

<details>
<summary>How to access your browser console (Click to expand)</summary>

# Chrome
- Press either CTRL + SHIFT + J to open the “console” tab of the Developer Tools.
- Alternative method:
    1. Press either CTRL + SHIFT + I or F12 to open the Developer Tools.
    2. Click the “console” tab.

# Safari
- Press CMD + ALT + I to open the Web Inspector.
- See Chrome’s step 2. (Chrome and Safari have pretty much identical dev tools.)

# IE9
1. Press F12 to open the developer tools.
2. Click the “console” tab.

# Firefox
- Press CTRL + SHIFT + K to open the Web console (COMMAND + SHIFT + K on Macs).
- or, if Firebug is installed (recommended):
    1. Press F12 to open Firebug.
    2. Click on the “console” tab.

# Opera
1. Press CTRL + SHIFT + I to open Dragonfly.
2. Click on the “console” tab.
</details>

**Comment by foxhoundv** [comment]
I know you want the log in this thread but it has my network information which I would rather not post here. I will send via email. Also I noticed DiCOM viewer wasn't wanting to display properly so I disabled it and I am still not able to view files and folders. I will send you two logs. Before and after DiCOM is disabled.

**Comment by szaimen** [comment]
So based on the logs you've send, it looks like the issue is caused by the maps app. Please try to disable it and check if it makes it work again!

**Comment by foxhoundv** [comment]
I disabled maps and it didn't work. I removed maps and it still doesn't work. I will send another log.

**Comment by szaimen** [comment]
Thanks! can you also try to disable the `files_mindmap` app?

**Comment by foxhoundv** [comment]
I disabled it and removed it and it still doesn't work.

**Comment by szaimen** [comment]
Hm... I still think that it is one of your installed apps. Can you try to disable all non-default apps and check if that makes it work? Afterwards you can enable them again one-for-one and see which one causes this...

**Comment by foxhoundv** [comment]
I think I disabled all non-default apps and it still doesn't work. Though I did disable "mail" and removed it and I am still getting errors for mail in the system log.

**Comment by szaimen** [comment]
Hm... Can you send all logs again when you try to access the files app. There should be something in the logs if it doesnt load...

Also can you post the updater logs?

Should be `sudo docker exec -it nextcloud-aio-nextcloud cat /var/www/html/data/update.log`

**Comment by foxhoundv** [comment]
![image](https://user-images.githubusercontent.com/16512250/211374306-f16bad82-8e8e-4354-8b3b-903b5a6bf3fb.png)
The first red section at the right is for the trashbin, the second red section is for the system tags, and the third is for onlyoffice.

**Comment by szaimen** [comment]
So something in here again?

`https://yourdomain.com/settings/admin/logging`

**Comment by szaimen** [comment]
And also are `extract`, `apporder` and `impersonate` disabled?

**Comment by szaimen** [comment]
Btw, the updater log looks good.

**Comment by foxhoundv** [comment]
> And also are `extract`, `apporder` and `impersonate` disabled?

All 3 are disabled and removed

**Comment by foxhoundv** [comment]
Whenever I refresh my page... there is an error that shows up after like 3 minutes...

`/apps/files/#:1 Uncaught (in promise) Error: A listener indicated an asynchronous response by returning true, but the message channel closed before a response was received`

**Comment by szaimen** [comment]
Thanks so it looks like this might be related to the external storage app. Can you check if the app is enabled and your storage is still working as expected?

**Comment by foxhoundv** [comment]
My external storage app was disabled. I enabled it and it shows that it connects fine to my network storage device.

**Comment by szaimen** [comment]
okay and it still does not show any files or folders? can you show the config of your external storage? (screenshot)?

**Comment by foxhoundv** [comment]
I still cannot access any files and folders through the browser.
![Screenshot 2023-01-09 112531](https://user-images.githubusercontent.com/16512250/211380512-45fcb001-cbe9-41b7-a9c9-05c955007cdf.png)

**Comment by szaimen** [comment]
Can you maybe try if it works in a different browser?

Also can you send all currently enabled apps?

`sudo docker exec -it --user www-data nextcloud-aio-nextcloud php occ app:list`

**Comment by foxhoundv** [comment]
I have tried accessing it through Edge, Chrome, and Brave...
Here is the list...

`WARNING: Error loading config file: /root/.docker/config.json: read /root/.docker/config.json: is a directory
Enabled:
  - activity: 2.17.0
  - admin_audit: 1.15.0
  - calendar: 4.2.1
  - cloud_federation_api: 1.8.0
  - comments: 1.15.0
  - contacts: 5.0.2
  - contactsinteraction: 1.6.0
  - dashboard: 7.5.0
  - dav: 1.24.0
  - federatedfilesharing: 1.15.0
  - federation: 1.15.0
  - files: 1.20.1
  - files_accesscontrol: 1.15.1
  - files_automatedtagging: 1.15.0
  - files_external: 1.17.0
  - files_pdfviewer: 2.6.0
  - files_retention: 1.14.1
  - files_trashbin: 1.15.0
  - files_versions: 1.18.0
  - firstrunwizard: 2.14.0
  - groupfolders: 13.1.0
  - logreader: 2.10.0
  - lookup_server_connector: 1.13.0
  - nextcloud-aio: 0.3.0
  - nextcloud_announcements: 1.14.0
  - notifications: 2.13.1
  - oauth2: 1.13.0
  - onlyoffice: 7.6.8
  - password_policy: 1.15.0
  - photos: 2.0.1
  - privacy: 1.9.0
  - provisioning_api: 1.15.0
  - recommendations: 1.4.0
  - related_resources: 1.0.3
  - serverinfo: 1.15.0
  - settings: 1.7.0
  - sharebymail: 1.15.0
  - spreed: 15.0.2
  - support: 1.8.0
  - survey_client: 1.13.0
  - systemtags: 1.15.0
  - text: 3.6.0
  - theming: 2.0.1
  - twofactor_backupcodes: 1.14.0
  - twofactor_totp: 7.0.0
  - user_status: 1.5.0
  - viewer: 1.9.0
  - weather_status: 1.5.0
  - workflowengine: 2.7.0
Disabled:
  - analytics: 4.6.0
  - announcementcenter: 6.4.0
  - bruteforcesettings
  - cfg_share_links: 3.0.2
  - circles: 25.0.0
  - deck: 1.8.3
  - encryption
  - external: 5.0.0
  - extract: 1.3.5
  - files_rightclick: 1.4.0
  - files_sharing: 1.17.0
  - flow_notifications: 1.5.0
  - forms: 3.0.3
  - impersonate: 1.11.0
  - notify_push: 0.5.0
  - passwords: 2022.12.21
  - passwords_handbook: 2022.12.21
  - side_menu: 3.5.1
  - suspicious_login
  - tasks: 0.14.5
  - unsplash: 2.1.1
  - user_ldap`

I also never had encryption enabled.

**Comment by szaimen** [comment]
Okay, seems to be caused by the onlyoffice app: https://github.com/ONLYOFFICE/onlyoffice-nextcloud/issues/745

**Reply by foxhoundv** [reply]
Does BorgBackup revert the containers and images to the previous ones before my update?

**Reply by szaimen** [reply]
It reverts all your data.

**Reply by foxhoundv** [reply]
What would you recommend to be the easiest way to go back to the previous version of AIO?

Also, how come in the updater log it says:
 - Keep legacy encryption enabled
 - Check encryption key format
But I never had it enabled?

**Reply by szaimen** [reply]
Simply restore the backup from before the update. This allows you to restore the working state.

**Reply by szaimen** [reply]
> Also, how come in the updater log it says:
> 
> * Keep legacy encryption enabled
> * Check encryption key format
>   But I never had it enabled?

this is a check that is always run and not a problem and also not related to your issue.

**Reply by foxhoundv** [reply]
It shows up in the enabled app list above, but when I go into my "Active Apps" setting there is no encryption app?

**Reply by szaimen** [reply]
yeah, the encryption app is disabled as you can see above... (it is listed below disabled...)

**Reply by szaimen** [reply]
![image](https://user-images.githubusercontent.com/42591237/211386051-396f31d8-8cac-42ff-880d-87bb29986394.png)

**Reply by foxhoundv** [reply]
Sorry I thought I read it under enabled.

**Reply by foxhoundv** [reply]
> Okay, seems to be caused by the onlyoffice app: [ONLYOFFICE/onlyoffice-nextcloud#745](https://github.com/ONLYOFFICE/onlyoffice-nextcloud/issues/745)

So I disabled onlyoffice and it still won't show files and folders. <-- I don't know if that helps since its part of the AIO though

**Reply by szaimen** [reply]
Hm, I am sorry but running out of ideas now. The files app works for me without any problem. Can you actually try to restore the backup, check if everything works and try the upgrade again?

**Reply by foxhoundv** [reply]
The restore failed...

'nextcloud_aio_nextcloud_data/tmp/
total: matches=0  hash_hits=0  false_alarms=0 data=1321397334

rsync[31] (sender) heap statistics:
  arena:      366526464   (bytes from sbrk)
  ordblks:         1401   (chunks not in use)
  smblks:             1
  hblks:              2   (chunks from mmap)
  hblkhd:        663552   (bytes from mmap)
  allmem:     367190016   (bytes from sbrk + mmap)
  usmblks:            0
  fsmblks:           96
  uordblks:     7342848   (bytes used)
  fordblks:   359183616   (bytes free)
  keepcost:      133712   (bytes in releasable chunk)

rsync[33] (server receiver) heap statistics:
  arena:      395423744   (bytes from sbrk)
  ordblks:         2544   (chunks not in use)
  smblks:             1
  hblks:              1   (chunks from mmap)
  hblkhd:        528384   (bytes from mmap)
  allmem:     395952128   (bytes from sbrk + mmap)
  usmblks:            0
  fsmblks:           96
  uordblks:     6938672   (bytes used)
  fordblks:   388485072   (bytes free)
  keepcost:      131472   (bytes in releasable chunk)

rsync[32] (server generator) heap statistics:
  arena:      210235392   (bytes from sbrk)
  ordblks:         4878   (chunks not in use)
  smblks:             1
  hblks:              1   (chunks from mmap)
  hblkhd:        528384   (bytes from mmap)
  allmem:     210763776   (bytes from sbrk + mmap)
  usmblks:            0
  fsmblks:           96
  uordblks:    10736848   (bytes used)
  fordblks:   199498544   (bytes free)
  keepcost:      394464   (bytes in releasable chunk)

Number of files: 212,968 (reg: 165,838, dir: 47,130)
Number of created files: 9,509 (reg: 7,828, dir: 1,681)
Number of deleted files: 5,604 (reg: 4,667, dir: 937)
Number of regular files transferred: 33,794
Total file size: 461.11G bytes
Total transferred file size: 1.32G bytes
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1333) [sender=3.2.3]
Literal data: 1.32G bytes
Matched data: 0 bytes
File list size: 2.29M
File list generation time: 0.008 seconds
File list transfer time: 0.000 seconds
Total bytes sent: 1.33G
Total bytes received: 18.03M

sent 1.33G bytes  received 18.03M bytes  9.45M bytes/sec
total size is 461.11G  speedup is 342.43
Something failed while restoring from backup.
'

**Reply by szaimen** [reply]
Can you send me the whole log via mail?

**Reply by szaimen** [reply]
Okay, this was the issue and is expected:
![image](https://user-images.githubusercontent.com/42591237/211394841-36563f3d-520b-4d91-87fe-3b3d7722f436.png)

So even though it says it failed, it should stil have restored the backup successfully. Does it make your instance work again? :)

**Comment by foxhoundv** [comment, marked as answer]
I did a backup before I updated. I started the containers and in the logs for Nextcloud it looks like it is updating.

**Reply by foxhoundv** [reply]
It worked!

It updated and now I see files and folders

**Reply by foxhoundv** [reply]
![image](https://user-images.githubusercontent.com/16512250/211397325-c76d4d21-14b7-46f8-8ea2-b8546822768e.png)
![image](https://user-images.githubusercontent.com/16512250/211397384-adfbf1cf-2d02-430e-ba01-693890105e42.png)

**Reply by szaimen** [reply]
Very strange but glad thaf it worked! :)

**Reply by foxhoundv** [reply]
Me too! Thanks for all your help!

**Reply by szaimen** [reply]
No problem!
